libs/wlgen/wlgen/rta.py

```diff
@@ -168,14 +168,18 @@ def calibrate(target):
                                  for key in pload) + "}")
 
         # Sanity check calibration values for big.LITTLE systems
-        if 'bl' in target.modules:
-            bcpu = target.bl.bigs_online[0]
-            lcpu = target.bl.littles_online[0]
-            if pload[bcpu] > pload[lcpu]:
-                log.warning('Calibration values reports big cores less '
-                            'capable than LITTLE cores')
-                raise RuntimeError('Calibration failed: try again or file a bug')
-            bigs_speedup = ((float(pload[lcpu]) / pload[bcpu]) - 1) * 100
+        if target.big_core:
+            bcpu = [i for i, t in enumerate(target.core_names) if t == target.big_core]
+            lcpu = [i for i, t in enumerate(target.core_names) if t == target.little_core]
+            for l in lcpu:
+                for b in bcpu:
+                    if pload[b] > pload[l]:
+                        log.warning('Calibration values reports big cores less'
+                                    ' capable than LITTLE cores')
+                        raise RuntimeError('Calibration failed: try again or '
```

Comment:
Could we just throw the exception with the message of log.warning ? that should be enough for debugging since we will get the exception message in the xUnit file generated by nosetest.